eth-contracts/contracts/service/DelegateManager.sol

```diff
@@ -306,20 +401,33 @@ contract DelegateManager is RegistryContract {
         emit Slash(_slashAddress, _amount, totalBalanceInStakingAfterSlash);
 
         uint totalDelegatedStakeDecrease = 0;
-
         // For each delegator and deployer, recalculate new value
         // newStakeAmount = newStakeAmount * (oldStakeAmount / totalBalancePreSlash)
         for (uint i = 0; i < spDelegateInfo[_slashAddress].delegators.length; i++) {
             address delegator = spDelegateInfo[_slashAddress].delegators[i];
             uint preSlashDelegateStake = delegateInfo[delegator][_slashAddress];
             uint newDelegateStake = (
-              totalBalanceInStakingAfterSlash.mul(preSlashDelegateStake)
+             totalBalanceInStakingAfterSlash.mul(preSlashDelegateStake)
             ).div(totalBalanceInStakingPreSlash);
             uint slashAmountForDelegator = preSlashDelegateStake.sub(newDelegateStake);
             delegateInfo[delegator][_slashAddress] -= (slashAmountForDelegator);
             delegatorStakeTotal[delegator] -= (slashAmountForDelegator);
             // Update total decrease amount
             totalDelegatedStakeDecrease += slashAmountForDelegator;
+            // Check for any locked up funds for this slashed delegator
```

Comment:
just thought of something else. maybe this makes it more complex and is unnecessary, but if you have enough unlocked funds to cover the slash, the slash could go through without cancelling your pending withdrawal. only if your balance isn't enough, we'd have to cancel this right?